tools/cache_creator/cache_creator.cpp

```diff
@@ -195,27 +252,13 @@ llvm::Expected<ElfLlpcCacheInfo> getElfLlpcCacheInfo(llvm::MemoryBufferRef elfBu
     if (notesError)
       return std::move(notesError);
 
-    Util::MetroHash::Hash hash = {};
-    bool foundHash = false;
-    uint32_t llpcVersion[2] = {};
-    bool foundLlpcVersion = false;
-
     for (const ElfFileTy::Elf_Note &note : noteRange) {
       llvm::StringRef noteName = note.getName();
-      llvm::ArrayRef<uint8_t> noteBlob = note.getDesc();
-
-      if (noteName == "AMD_llpc_cache_hash") {
-        assert(noteBlob.size() == sizeof(Util::MetroHash::Hash) && "Invalid AMD_llpc_cache_hash note");
-        memcpy(hash.bytes, noteBlob.data(), sizeof(hash));
-        foundHash = true;
-      } else if (noteName == "AMD_llpc_version") {
-        assert(noteBlob.size() == sizeof(llpcVersion) && "Invalid AMD_llpc_version note");
-        memcpy(llpcVersion, noteBlob.data(), sizeof(llpcVersion));
-        foundLlpcVersion = true;
-      }
+      llvm::StringRef noteBlob = note.getDescAsStringRef();
 
-      if (foundHash && foundLlpcVersion)
-        return ElfLlpcCacheInfo{hash, llpcVersion[0], llpcVersion[1]};
+      if (noteName == "AMDGPU") {
```

Comment:
I think we can remove the string comparision by checking the note type.  The note type should be `ELF::NT_AMDGPU_METADATA`.  You can see it used in llvm-project/llvm/lib/Target/AMDGPU/MCTargetDesc/AMDGPUTargetStreamer.cpp.